### PR TITLE
Update tests for Categories, Link, SearchPage and UserProfileEdit

### LINKED ISSUE #10743: Remove export of mapStateToProps from some components (part A)
The fix for #10140 will be quite lengthy, with a number of test suites needing to be updated, so I think it makes sense to do it in chunks, which will also make reviewing the PRs a bit less onerous.

This issue with start by dealing with:
- src/amo/components/Categories/index.js
- src/amo/components/Link/index.js
- src/amo/pages/SearchPage/index.js
- src/amo/pages/SearchTools/index.js
- src/amo/pages/UserProfileEdit/index.js

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #10743

### CODE REVIEW

File: tests/unit/amo/components/TestCategories.js [hunk 1]
```diff
@@ -129,21 +132,27 @@ describe(__filename, () => {
...
-      addonType: ADDON_TYPE_EXTENSION,
-      loading: true,
-    });
+    store.dispatch(
+      fetchCategories({
+        errorHandlerId: createStubErrorHandler().id,
+      }),
+    );
+
+    const root = render({ addonType: ADDON_TYPE_EXTENSION });
```
Comment (eviljeff) [inline tests/unit/amo/components/TestCategories.js:141]: no `loading` now?

File: tests/unit/amo/components/TestCategories.js [hunk 2]
```diff
@@ -129,21 +132,27 @@ describe(__filename, () => {
...
```
Comment (eviljeff) [inline tests/unit/amo/components/TestCategories.js:155]: same question

**Review by eviljeff — APPROVED** [review]
I don't see anything glaringly wrong but I'm really not familiar with shallow rendering and how `shallowUntilTarget` might differ in edge cases, so might need a second opinion.

File: tests/unit/amo/components/TestCategories.js (hunk 1 above)
Comment (bobsilverberg) [reply tests/unit/amo/components/TestCategories.js:141]: Passing `loading` as a property was never really the right way to do it. Ideally, if you want a component to be in a loading state, you dispatch an action that will put it into that state (which is what I've done above with `store.dispatch(fetchCategories(...))`). Most of this patch is either removing tests that are no longer needed, or replacing tests that explicitly tested `mapStateToProps` with tests that test the functionality of `mapStateToProps` without explicitly calling it. This is an example of the latter. In this case it is `mapStateToProps` that sets `loading`, which is based on the current Redux state, which I set by dispatching `fetchCategories`.

File: tests/unit/amo/components/TestCategories.js (hunk 2 above)
Comment (bobsilverberg) [reply tests/unit/amo/components/TestCategories.js:155]: See above 😄

**Comment by bobsilverberg** [comment]
> I don't see anything glaringly wrong but I'm really not familiar with shallow rendering and how `shallowUntilTarget` might differ in edge cases, so might need a second opinion.

There really shouldn't be much difference between the basic `shallow` and `shallowUntilTarget`. The latter allows for better interaction with the Redux store, which is what we're looking for here. Unlike the other test suites in this patch, the one for Categories was quite old and really needed to be updated. If your only concern is that you might have missed something in the switch from `shallow` to `shallowUntilTarget`, then I think we're fine.
